commit f9af8d0f3158019fe8cead162c2e39167705865b
Author: John Johansen <john.johansen@canonical.com>
Date:   Mon Aug 22 14:14:48 2016 -0700

    UBUNTU: SAUCE: apparmor: fix vec_unique for vectors larger than 8
    
    the vec_unique path for large vectors is broken, leading to oopses
    when a file handle is shared between 8 different security domains, and
    then a profile replacement/removal causing a label invalidation (ie. not
    all replacements) is done.
    
    BugLink: http://bugs.launchpad.net/bugs/1579135
    
    Signed-off-by: John Johansen <john.johansen@canonical.com>
    Signed-off-by: Leann Ogasawara <leann.ogasawara@canonical.com>

diff --git a/security/apparmor/label.c b/security/apparmor/label.c
index 144d759..c11ca99 100644
--- a/security/apparmor/label.c
+++ b/security/apparmor/label.c
@@ -229,7 +229,7 @@ static inline int unique(struct aa_profile **vec, int n)
 	AA_BUG(!vec);
 
 	pos = 0;
-	for (i = 1; 1 < n; i++) {
+	for (i = 1; i < n; i++) {
 		int res = profile_cmp(vec[pos], vec[i]);
 		AA_BUG(res > 0, "vec not sorted");
 		if (res == 0) {
